pkg/collector/corechecks/net/networkv2/network_windows.go

```diff
@@ -0,0 +1,352 @@
+	}
+
+	tcp4Stats, err := c.net.TcpStats("tcp4")
+	if err != nil {
+		return err
+	}
+
+	tcp6Stats, err := c.net.TcpStats("tcp6")
+	if err != nil {
+		return err
+	}
+
+	// Create tcp metrics that are a sum of tcp4 and tcp6 metrics
+	if !reflect.ValueOf(tcp4Stats).IsZero() && !reflect.ValueOf(tcp6Stats).IsZero() {
+		tcpAllStats := &mibTcpStats{
+			DwRtoAlgorithm: tcp4Stats.DwRtoAlgorithm + tcp6Stats.DwRtoAlgorithm,
+			DwRtoMin:       tcp4Stats.DwRtoMin + tcp6Stats.DwRtoMin,
+			DwRtoMax:       tcp4Stats.DwRtoMax + tcp6Stats.DwRtoMax,
+			DwMaxConn:      tcp4Stats.DwMaxConn + tcp6Stats.DwMaxConn,
+			DwActiveOpens:  tcp4Stats.DwActiveOpens + tcp6Stats.DwActiveOpens,
+			DwPassiveOpens: tcp4Stats.DwPassiveOpens + tcp6Stats.DwPassiveOpens,
+			DwAttemptFails: tcp4Stats.DwAttemptFails + tcp6Stats.DwAttemptFails,
+			DwEstabResets:  tcp4Stats.DwEstabResets + tcp6Stats.DwEstabResets,
+			DwCurrEstab:    tcp4Stats.DwCurrEstab + tcp6Stats.DwCurrEstab,
+			DwInSegs:       tcp4Stats.DwInSegs + tcp6Stats.DwInSegs,
+			DwOutSegs:      tcp4Stats.DwOutSegs + tcp6Stats.DwOutSegs,
+			DwRetransSegs:  tcp4Stats.DwRetransSegs + tcp6Stats.DwRetransSegs,
+			DwInErrs:       tcp4Stats.DwInErrs + tcp6Stats.DwInErrs,
+			DwOutRsts:      tcp4Stats.DwOutRsts + tcp6Stats.DwOutRsts,
+			DwNumConns:     tcp4Stats.DwNumConns + tcp6Stats.DwNumConns,
+		}
+		submitMetricsFromStruct(sender, "system.net.tcp", tcpAllStats, tcpStatsMapping)
+	}
+
+	if !reflect.ValueOf(tcp4Stats).IsZero() {
+		submitMetricsFromStruct(sender, "system.net.tcp4", tcp4Stats, tcpStatsMapping)
+	}
+	if !reflect.ValueOf(tcp6Stats).IsZero() {
+		submitMetricsFromStruct(sender, "system.net.tcp6", tcp6Stats, tcpStatsMapping)
+	}
+
+	return nil
+}
+
+func submitMetricsFromStruct(sender sender.Sender, metricPrefix string, tcpStats *mibTcpStats, tcpStatsMapping map[string]string) {
+	s := reflect.ValueOf(tcpStats).Elem()
+	sType := s.Type()
+	for i := 0; i < s.NumField(); i++ {
+		fieldName := sType.Field(i).Name
+		metricName := metricPrefix + tcpStatsMapping[fieldName]
+		metricValue := s.Field(i).Uint()
+		if strings.HasSuffix(metricName, ".connections") || strings.HasSuffix(metricName, ".current_established") {
+			sender.Gauge(metricName, float64(metricValue), "", nil)
+		} else {
+			sender.Rate(metricName, float64(metricValue), "", nil)
+			sender.MonotonicCount(fmt.Sprintf("%s.count", metricName), float64(metricValue), "", nil)
+		}
+	}
+}
+
+func (c *NetworkCheck) isDeviceExcluded(deviceName string) bool {
+	for _, excludedDevice := range c.config.instance.ExcludedInterfaces {
+		if deviceName == excludedDevice {
+			return true
+		}
+	}
+	if c.config.instance.ExcludedInterfacePattern != nil {
+		return c.config.instance.ExcludedInterfacePattern.MatchString(deviceName)
+	}
+	return false
+}
+
+func submitInterfaceMetrics(sender sender.Sender, interfaceIO net.IOCountersStat) {
+	tags := []string{fmt.Sprintf("device:%s", interfaceIO.Name), fmt.Sprintf("device_name:%s", interfaceIO.Name)}
```

Comment:
`device_name` seems to have been deprecated in the Python version ?

Reply:
hmm this is a break in behavior compatibility with the current iot network check then?

Reply:
it looks like this is for compatibility with some container checks: https://github.com/DataDog/datadog-agent/pull/5412

I discussed with Srdjan in our 1:1 and I think we need to keep it for the linux check but since it isn't there for the windows one I will remove it 👍

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/37240/commits/a8f06404638e4898f0962b431c7e75a65bd21ed9#diff-f8bfa24f1e3f0888eca542771acc5c5a2fa394dacf5e279c0c0955c8d97fdb41L302